stable/atlantis/templates/atlantis-consul-template.yaml

```diff
@@ -0,0 +1,26 @@
+{{- if .Values.vault }}
```

Comment:
I don't think this vault stuff should be in the atlantis chart. It's too specific to your use-case, most people won't need this.

Reply:
If that is the case, what is the criteria to be met for adding in additional functionality for things like this in Atlantis? There is already similar blocks within the templates doing the same function for other providers, and other source code repositories, so to an extent this follows that convention. I think think the assumption is also being made that secrets management integration for Atlantis should be handled outside of the Atlantis chart?

Reply:
You're basically saying most people don't need secrets management.

Reply:
@lkysow 
How do you expect most people will keep their passwords, tokens, certs, etc. out of source control without integrating with some sort of secrets management? It makes a lot of sense to me to support using Terraform and Vault together if you're already using both Hashicorp projects. 

Sure, you could manually drop in the Vault pieces or deploy Atlantis from static manifests, but you lose the benefits of package management offered by helm.

Reply:
> If that is the case, what is the criteria to be met for adding in additional functionality for things like this in Atlantis?

My criteria for judging if additional functionality to be added to this chart is whether the functionality augments the existing chart using kubernetes primitives. If the additional functionality requires new binaries to run, new software, ex. the vault agent and consul-template, then I would recommend users who need that to fork the chart. Forking a helm chart isn't a big issue unlike forking an application. Once your helm chart is deploying Atlantis the way you like, you're unlikely to touch it again. Any updates to the chart in this repo are going to be small updates, adding configuration knobs that people need. 

> You're basically saying most people don't need secrets management.

I am not saying this. There are many ways to add secrets to Atlantis that are already exposed in this chart, for example via Kubernetes secrets that expose environment variables for certain providers. I'm saying that the tradeoff in complexity adding this vault method of secret management isn't worth it. Any addition to this chart needs to be fully tested and maintained forever.

> How do you expect most people will keep their passwords, tokens, certs, etc. out of source control without integrating with some sort of secrets management?

As I mentioned, there are other mechanisms.

> It makes a lot of sense to me to support using Terraform and Vault together if you're already using both Hashicorp projects.

Totally, I just don't want to support that in this helm chart because I don't think the complexity tradeoff is worth it. Nothing is stopping you from doing this yourself.

> Sure, you could manually drop in the Vault pieces or deploy Atlantis from static manifests, but you lose the benefits of package management offered by helm.

You can still use helm. You can copy the `stable/atlantis` folder to your own repo with the changes you want and then instead of running `helm install stable/atlantis` you run `helm install ./stable/atlantis`. You can then fully enjoy the helm package management with your custom chart and as I talk about above, you're not going to miss out on any important updates to this chart because I don't forsee it changing much in the future.


I know that you're eager to add this functionality to the helm chart and I thank you for all your hard work on it, however for the reasons outlined above, I don't wish to add these changes to the chart.

Reply:
Luke, thanks for your well thought out and explanatory response. I want to make sure you understand the whole premise of where this is coming from:
```


    You're basically saying most people don't need secrets management.

I am not saying this. There are many ways to add secrets to Atlantis that are already exposed in this chart, for example via Kubernetes secrets that expose environment variables for certain providers. I'm saying that the tradeoff in complexity adding this vault method of secret management isn't worth it. Any addition to this chart needs to be fully tested and maintained forever.

    How do you expect most people will keep their passwords, tokens, certs, etc. out of source control without integrating with some sort of secrets management?

As I mentioned, there are other mechanisms.
```
Based on your response, I'm not getting the answer of how the Atlantis or people using it are handling resource creation via Terraform where credentials need to be passed for specific resource creation. This is specifically NOT via the cloud provider secrets you commented on. Lets say that via a .tf file you create a database and a VM and you need to provide credentials for accounts on the machines. With the vault provider piece added in, this problem is figured out. At the moment with the project, it appears that this has been overlooked or perhaps you just have not commented on a prescribed way to handle this. I'm happy to maintain a fork (especially if the project isn't slated to change a lot from here forward) but this seems like a pretty table-stakes issue that isn't being addressed.

Reply:
Thanks for the reply @lkysow. I get your point about maintaining complexity in the long term. That's part of the motivation for this PR. 

I do see it as a shortcoming of this chart that it doesn't provide a way for Atlantis users to keep these types of secrets out of clear-text in SCM out of the box, but I respect your position/decision as a maintainer. 

@jhauer you should probably close this PR.

Reply:
Closing out the PR at this point.

Reply:
Ahh I see, yes that's definitely a problem. I think users are often using the aws_ssm_parameter store or something similar, or they're using git-crypt, or other mechanisms. If there are more users using vault in this manner and they have issues then hopefully they find this PR and put a 👍 on it and we can revisit whether it makes sense to support this in the stable chart.